phd-tests/tests/src/disk.rs

```diff
@@ -4,60 +4,136 @@
 
 use phd_framework::{
     disk::{fat::FatFilesystem, DiskSource},
-    test_vm::{DiskBackend, DiskInterface},
+    guest_os::GuestOsKind,
+    test_vm::{DiskBackend, DiskInterface, MigrationTimeout},
+    TestVm,
 };
 use phd_testcase::*;
-use tracing::{info, warn};
+use uuid::Uuid;
 
-#[phd_testcase]
-async fn in_memory_backend_smoke_test(ctx: &Framework) {
-    const HELLO_MSG: &str = "hello oxide!";
-
-    let mut cfg = ctx.vm_config_builder("in_memory_backend_test");
-    let mut data = FatFilesystem::new();
-    data.add_file_from_str("hello_oxide.txt", HELLO_MSG)?;
+/// Creates a VM with an in-memory disk backed by the supplied `data`, waits for
+/// it to boot, and issues some shell commands to find the
+///
+/// Returns a tuple containing the created VM and the path to the guest disk
+/// device representing the in-memory disk.
+async fn launch_vm_and_find_in_memory_disk(
+    ctx: &Framework,
+    vm_name: &str,
+    data: DiskSource<'_>,
+    readonly: bool,
+) -> anyhow::Result<(TestVm, String)> {
+    let mut cfg = ctx.vm_config_builder(vm_name);
     cfg.data_disk(
         "data-disk-0",
-        DiskSource::FatFilesystem(data),
+        data,
         DiskInterface::Virtio,
-        DiskBackend::InMemory { readonly: true },
+        DiskBackend::InMemory { readonly },
         24,
     );
-
     let mut vm = ctx.spawn_vm(&cfg, None).await?;
-    if vm.guest_os_kind().is_windows() {
-        phd_skip!("this test uses mount options not supported by Cygwin");
-    }
     vm.launch().await?;
     vm.wait_to_boot().await?;
 
-    // Some guests expose a /dev/disk/by-path directory that contains symlinks
-    // mapping PCI paths to the underlying device nodes under /dev. If this is
-    // present, check the mapping for a virtio disk at 0.24.0.
-    //
-    // The commands after this try to mount the in-memory disk and assume that
-    // its device is located at /dev/vda. If the by-path directory is present,
-    // try to check that the disk is located there and fail the test early if
-    // it's not. If the by-path directory is missing, put up a warning and hope
-    // for the best.
-    let dev_disk = vm.run_shell_command("ls /dev/disk").await?;
-    if dev_disk.contains("by-path") {
-        let ls = vm.run_shell_command("ls -la /dev/disk/by-path").await?;
-        info!(%ls, "guest disk device paths");
-        assert!(ls.contains("virtio-pci-0000:00:18.0 -> ../../vda"));
+    let device_path = if let Some(win_vm) = vm.get_windows_vm() {
```

Comment:
tiny but nice: if you added an `impl<'a> Deref for WindowsVm<'a> { type Target = TestVm<'a>; }` you could write this as `if let Some(vm) = vm.as_windows_vm()` and not have to know which "vm" is the right one for what you want to call. it'd all be `vm.run_powershell_command` and `vm.run_shell_command` in this arm.